test/behaviour/resolution/framework/common/RuleResolutionBuilder.java

```diff
@@ -119,34 +125,52 @@
         return Graql.and(result);
     }
 
-    public LinkedHashMap<String, Statement> statementToResolutionProperties(Statement statement) {
+    public LinkedHashMap<String, Statement> statementToResolutionProperties(final Transaction tx, Statement statement, final Boolean inferred) {
```

Comment:
why is this a Boolean instead of a boolean? Do we need a null on top of true/false?

Reply:
yes - null is passed in as an argument in some places.

Reply:
if you need to you can do `my_method(@Nullable Boolean my_arg_that_can_be_null)`